src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -146,9 +146,20 @@ private function parseDefinitions($content, $file)
         if (!is_array($content['services'])) {
             throw new InvalidArgumentException(sprintf('The "services" key should contain an array in %s. Check your YAML syntax.', $file));
         }
+        $defaults = isset($content['service_defaults']) ? $content['service_defaults'] : array();
+        $defaultKeys = array('public', 'tags', 'autowire');
+
+        if (!is_array($defaults)) {
+            throw new InvalidArgumentException(sprintf('Service defaults must be an array, "%s" given in "%s".', gettype($defaults), $file));
+        }
+        foreach ($defaults as $key => $default) {
+            if (!in_array($key, $defaultKeys)) {
```

Comment:
done this way for consistency with the many other in_array in this file, good enough anyway